src/set_complete.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2022 Collabora Limited
+# Author: Jeny Sadadia <jeny.sadadia@collabora.com>
+
+import logging
+import os
+import sys
+
+import kernelci
+import kernelci.config
+import kernelci.db
+from kernelci.cli import Args, Command, parse_opts
+
+from logger import Logger
+
+
+class SetComplete:
+
+    def __init__(self, configs, args):
+        self._db_config = configs['db_configs'][args.db_config]
+        api_token = os.getenv('API_TOKEN')
+        self._db = kernelci.db.get_db(self._db_config, api_token)
+        self._logger = Logger("config/logger.conf", "set_complete")
+
+    def run(self):
+        sub_id = self._db.subscribe_node_channel(filters={
+            'status': ('complete', 'timeout'),
+        })
+        self._logger.log_message(logging.INFO, "Listening for node events... ")
+        self._logger.log_message(logging.INFO, "Press Ctrl-C to stop.")
+
+        try:
+            while True:
+                node = self._db.receive_node(sub_id)
+                if node['parent'] is None:
+                    continue
+                nodes = self._db.get_nodes({"parent": node['parent']})
+                ret = False
+                for node in nodes:
+                    if node['status'] == "pending":
+                        ret = True
+                        break
+                if ret:
+                    continue
+                root_node = self._db.get_root_node(node['_id'])
```

Comment:
Nope, we get a complete node in event and get sibling nodes from it. Parent node will be set to complete rather than a root node. Added a few tweaks for it.

Reply:
Ah yes. I think the variable names are a bit ambiguous e.g. `node` and `nodes` could be anything.

Also `get_root_node()` is basically `get_node(node['parent'])`, maybe we don't need a special method for that.

Reply:
> I think the variable names are a bit ambiguous

Ah I see you've already updated that, thanks